## Условие 1: 1 задача
- Скачать файл в уроке 'kc_house_data.csv'.
- Считать данные с помощью pandas
- Вывести на экран первые 5 строк
- Посмотреть на описание признаков и на их содержание

In [19]:
import pandas as pd
df=pd.read_csv('kc_house_data.csv', sep=',')
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

Посмотреть на описание признаков и на их содержание:
- id - идентификатор
-date - дата
-price - цена
-bedrooms - количество спален
-bathrooms - количество ванных комнат
-sqft_living - кв.метры жилые
-sqft_lot - кв.метры общие
-floors - количество этажей
-waterfront - признак набережной
-view - признак хорошего вида из окон
-condition - состояние
-grade - оценка
-sqft_above - площадь этажей
-sqft_basement-  площадь подвала
-yr_built - год постройки
-yr_renovated -  год ремонта
-zipcode - почтовый индекс
-lat - широта
-long - долгота
-sqft_living15 -  кв. метры жилой площади у 15 соседей
-sqft_lot15 - кв. метры общей площади у 15 соседей

## Условие 2: 2 задача
- Проведите первичный анализ данных
- Изучите типы данных
- Найдите количество пропущенных ячеек в данных
- Посчитайте основные статистики по всем признакам и поизучайте их
- Пишите выводы

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [21]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [22]:
df.describe(include=['object'])

date
count             21613
unique              372
top     20140623T000000
freq                142

In [ ]:
df.shape

(21613, 21)

In [23]:
df.isnull().sum()

id               0
date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
view             0
condition        0
grade            0
sqft_above       0
sqft_basement    0
yr_built         0
yr_renovated     0
zipcode          0
lat              0
long             0
sqft_living15    0
sqft_lot15       0
dtype: int64

Выводы: <br>
 Представленные данные представляют таблицу, состояющую из 21613 строк и 21 столбца. Пропущенных ячеек в таблице нет. Типы данных: float64(5), int64(15), object(1)

## Условие 3: 3 задача
### Ответьте на несколько вопросов:


3.1. В каком диапазоне изменяются стоимости недвижимости?


In [27]:
df['price'].min(), df['price'].max()

(75000.0, 7700000.0)

3.2. Какую долю в среднем занимают жилая площадь от всей площади по всем домам?


In [29]:
df.sqft_living.mean() * 100/df.sqft_lot.mean()

13.76781757959227

3.3. Как много домов с разными этажами в данных?


In [30]:
df['floors'].value_counts()

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

3.4. Насколько хорошие состояния у домов в данных?


In [31]:
df['condition'].value_counts()

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

3.5. Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [33]:
print(f'Самый старый дом: {df.yr_built.min()}\nСамый новый дом: {df.yr_built.max()}')

Самый старый дом: 1900
Самый новый дом: 2015


## Условие 4: 4 задача
### Ответьте на несколько вопросов

4.1. Сколько в среднем стоят дома, у которых 2 спальни?

In [34]:
df[df['bedrooms'] == 2]['price'].mean()

401372.681884058

4.2. Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [35]:
df[df['price'] > 600000]['sqft_lot'].mean()

20442.524776214832

4.3. Как много домов коснулся ремонт?

In [40]:
df[df['yr_renovated'] != 0].shape[0]

914

4.4. Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [37]:
df[df['grade'] > 10]['price'].mean() - df[df['grade'] < 4]['price'].mean()

1488885.1175298805

## Условие 5: 5 задача
### Ответьте на несколько вопросов


5.1. Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

In [38]:
df[(df['waterfront'] ==1) & (df['bathrooms']>=3) & (df['sqft_basement']>0)].shape[0]

41

5.2. Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

In [42]:
df[((df['view'] == 4) | (df['waterfront'] == 1)) & (df['condition'] == 5) & (df['yr_built'] >= 1980)]['price'].min()

1295000.0

In [43]:
df[((df['view'] == 4) | (df['waterfront'] == 1)) & (df['condition'] == 5) & (df['yr_built'] >= 1980)]['price'].max()

3000000.0

5.3. Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [45]:
df[(df['sqft_basement'] == 0) & (df['floors'] == 2) & (df['price'] < 150000)]['condition'].mean()

2.8333333333333335